.travis.yml

```diff
@@ -3,6 +3,9 @@ language: cpp
 sudo: required
 dist: trusty
 
+git:
+  depth: 1
```

Comment:
I also noticed it, I set it to 1 to,
* speeding up failing old commit in the queue
* possibly saving some time when doing `git clone`

I think it's OK to fail old if there's a new commit pushed to an queued branch. Because the resource is limited, quickly fail building an outdated commit can make room for the latest commit. I think that's why [Building only the latest commit](https://docs.travis-ci.com/user/customizing-the-build#Building-only-the-latest-commit) is introduced.

As far as I know, `depth 1` and `Building only the latest commit` is different. The former may cause the running job fail for being unable to find the commit id, while the latter will not fail if the commit id is within 50 depth.

As [Git Clone Depth](https://docs.travis-ci.com/user/customizing-the-build#Git-Clone-Depth) suggested,
> Travis CI clones repositories to a depth of 50 commits, which is only really useful if you are performing git operations.

I also did an experiment, which shows that this will not affect two branches.